CorsixTH/Lua/world.lua

```diff
@@ -2279,19 +2279,7 @@ function World:getRoomNameAndRequiredStaffName(room_id)
   for key, _ in pairs(required_staff) do
     staff_name = key
   end
-  required_staff = staff_name -- This is the "programmatic" name of the staff.
-  if staff_name == "Nurse" then
-    staff_name = _S.staff_title.nurse
-  elseif staff_name == "Psychiatrist" then
-    staff_name = _S.staff_title.psychiatrist
-  elseif staff_name == "Researcher" then
-    staff_name = _S.staff_title.researcher
-  elseif staff_name == "Surgeon" then
-    staff_name = _S.staff_title.surgeon
-  elseif staff_name == "Doctor" then
-    staff_name = _S.staff_title.doctor
-  end
-  return room_name, required_staff, staff_name
+  return room_name, staff_name, translateStaffclass(staff_name)
```

Comment:
The problem is your capitalisation :)
``translateStaffClass`` not ``translateStaffclass``